IMEsec-USP/container-manager
# Transforms container-manager into a restful API

*It's better to read this PR commit by commit for less code at a time.*

### What does this code do?
Using [martini](https://github.com/go-martini/martini) as a HTTP handler and [zerolog](https://github.com/rs/zerolog) (although not using the zero-allocations feature, at least for now) as a logger, create a restful API that executes the remote procedures that we were expecting container-manager to do.

### Why?

We're going to call container-manager via an API from ada. Therefore, we need this change.
Also, an explanation on why I separated the apis for "pull" and "restart" -- so that we can restart
containers without logging into our servers via ada and so that container-manager already pulls the image for the 

### How do I test it?

1. Spin up the `docker-compose up --build -d` from the `test-containers` folder.
2. `curl -v localhost:3000/_healthcheck` should return 200.
3. `curl -v localhost:3000/restart/lh-timer` should restart lh-timer and the tests from `test.sh` should not fail.
4. `curl -v localhost:3000/pull/lh-timer` should pull the image for LH timer.
5. `docker-compose logs` should have some movement of what happened.

### What's next?

- [ ] Load applications using [viper](https://github.com/spf13/viper) instead of hardcoding them in code. If possible, load it from a configuration coming from docker-compose.

- [ ] Load production values and make ada use this API!

- [ ] Figure out a way to deploy container-manager by restarting itself.
